pkg/epoch/epoch_test.go

```diff
@@ -0,0 +1,77 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package epoch
+
+import (
+	"os"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func rightAfter(t1, t2 time.Time) bool {
+	if runtime.GOOS == "windows" {
+		// Low timer resolution on Windows
+		return (t2.After(t1) && t2.Before(t1.Add(100*time.Millisecond))) || t2.Equal(t1)
+	}
+	return t2.After(t1) && t2.Before(t1.Add(10*time.Millisecond))
+}
+
+func TestSourceDateEpoch(t *testing.T) {
+	if _, ok := os.LookupEnv(SourceDateEpochEnv); ok {
+		t.Skipf("%s is already set", SourceDateEpochEnv)
+	}
```

Comment:
Is this worth skipping or can we unset and restore the value?

```suggestion
	if s, ok := os.LookupEnv(SourceDateEpochEnv); ok {
		os.Unsetenv(SourceDateEpochEnv)
		t.Cleanup(func() {os.Setenv(SourceDateEpochEnv, s)})
	}
```

Reply:
Fixed